src/App.tsx

```diff
@@ -1,27 +1,58 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  // eslint-disable-next-line
-  console.log(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  clockName: number;
+  isClockVisible: boolean
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    clockName: 1,
+    isClockVisible: true,
+  };
+
+  generateClockName = () => {
+    const randomNumber = Math.round(Math.random() * 1000);
+
+    this.setState({ clockName: randomNumber });
+  };
+
+  hideClock = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  showClock = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  render() {
+    const { clockName } = this.state;
+
+    return (
+      <div className="App">
+        <div className="App__wrapper">
+          {this.state.isClockVisible
+            && <Clock data-cy="time" name={clockName} />}
```

Comment:
maybe you can take this to the bottom, that way the buttons will not jump

Reply:
done